packages/waku/src/router/base-types.ts

```diff
@@ -0,0 +1,22 @@
+import type { createPages } from './create-pages.js';
```

Comment:
I'm not comfortable having this dependency in this "base" file.

Reply:
removed this dependency with moving RouterSpec and AnyPage to `inferred-router-types.ts`and moved `InferredPaths` to a new file: `types-from-route-config.ts`

I'll move these to `src/router/create-pages-utils/*.ts` per your other thread next, but this change should address the concern of the dep on create-pages